processor/cloningfanoutconnector.go

```diff
@@ -35,37 +38,44 @@ import (
 
 // CreateMetricsCloningFanOutConnector is a placeholder function for now.
 // It supposed to create an old type connector or a new type connector based on type of provided metrics consumer.
-func CreateMetricsCloningFanOutConnector(baseMetricsConsumers []consumer.MetricsConsumerBase) consumer.MetricsConsumerOld {
-	// TODO: CreateMetricsCloningFanOutConnector doesn't support new type of consumers
-	// until internal data structure provides Clone method.
-	metricsConsumers := make([]consumer.MetricsConsumerOld, 0, len(baseMetricsConsumers))
-	for _, baseMetricsConsumer := range baseMetricsConsumers {
-		metricsConsumer, ok := baseMetricsConsumer.(consumer.MetricsConsumerOld)
-		if !ok {
-			panic("CreateMetricsCloningFanOutConnector does not support new type of MetricsConsumer")
+func CreateMetricsCloningFanOutConnector(mcs []consumer.MetricsConsumerBase) consumer.MetricsConsumerBase {
+	metricsConsumersOld := make([]consumer.MetricsConsumerOld, 0, len(mcs))
+	metricsConsumers := make([]consumer.MetricsConsumer, 0, len(mcs))
+	allMetricsConsumersOld := true
+	for _, mc := range mcs {
+		if metricsConsumer, ok := mc.(consumer.MetricsConsumer); ok {
```

Comment:
nit: because if/else and another type switch with a local defined on `switch` probably will look simpler

Reply:
I personally like more if/else

Reply:
This condition is only for the transition state. So will be removed soon anyway :)

Reply:
actually `switch` will not give us right result. we need only one behavior even if `ms` implements both interfaces, and that's what `if` provides